src/dotnet/commands/dotnet-publish/PublishCommand.cs

```diff
@@ -272,33 +316,60 @@ private static void PublishRefs(LibraryExport export, string outputPath, bool de
             }
         }
 
-        private static int PublishHost(ProjectContext context, string outputPath, CommonCompilerOptions compilationOptions)
+        private static int RenamePublishedHost(ProjectContext context, string outputPath, CommonCompilerOptions compilationOptions)
         {
             if (context.TargetFramework.IsDesktop())
             {
                 return 0;
             }
 
-            foreach (var binaryName in Constants.HostBinaryNames)
+            var publishedHostFile = ResolvePublishedHostFile(outputPath);
+            if (publishedHostFile == null)
             {
-                var hostBinaryPath = Path.Combine(AppContext.BaseDirectory, binaryName);
-                if (!File.Exists(hostBinaryPath))
-                {
-                    Reporter.Error.WriteLine($"Cannot find {binaryName} in the dotnet directory.".Red());
-                    return 1;
-                }
+                Reporter.Output.WriteLine($"publish: warning: host executable not available in dependencies, using host for current platform");
```

Comment:
I agree, that this should be a host. 

The host is not random however, it is copied from the shared fx directory (falls back to previous behavior). 

This will be a breaking change to remove the fallback, so I'm not too sure the best way to go about this.